psycop/common/model_training_v2/problem_type/problem_type_base.py

```diff
@@ -17,5 +20,5 @@ def train(self, x: PolarsFrame, y: PolarsFrame) -> None:
     def evaluate(self, x: PolarsFrame, y: PolarsFrame) -> TrainingResult:
         ...
 
-    def predict_proba(self, x: PolarsFrame) -> pl.Series:
+    def predict_proba(self, x: PolarsFrame) -> pd.Series[float]:
```

Comment:
Unsure we can make this assumption with the return type. Multi-label classification will return an array/dataframe with the same number of columns as the number of output labels. We've already considered the multilabel case in the .train and .evaluate methods of ProblemType (by using PolarsFrame instead of Series for y), so I think we should try to come up with a smart solution here as well

Reply:
We actually already made a `PredProbaSeries` type which we use in BinaryClassification. We are never directly calling .predict_proba -- it's called via `.train` -- so one solution could be to make it private, have it return a PredProbaSeries (where each row could be a list of floats for the multilabel case) and handle the rest in the `.train` method of the ProblemType. Let's talk about it

Reply:
That sounds like a very good solution. Referencing this in #392.